runtime/syntax/c.vim

```diff
@@ -277,6 +306,13 @@ if !exists("c_no_c99") " ISO C99
   syn keyword	cType		intptr_t uintptr_t
   syn keyword	cType		intmax_t uintmax_t
 endif
+if !exists("c_no_c23")
+  syn keyword	cOperator	typeof typeof_unqual
+  syn keyword	cType           _BitInt _Decimal32 _Decimal64 _Decimal128
```

Comment:
Maybe I'm misreading the conditions, but does this make these enabled in C++ too? These keywords are defined in the C23 standard, but are not in any C++ standard (although `typeof` and the `_DecimalNN` types are supported by G++ as extensions). It won't hurt to treat `_BitInt` and the decimal types as keywords in C++ code, since they're reserved names and so can't be used for users' identifiers. But in theory a C++ program could use `typeof` and `typeof_unqual` as variable names, or functions, or macros, but this would highlight them as keywords (... unless I'm just reading it wrong).